tests/testthat/test-batch-chat.R

```diff
@@ -0,0 +1,125 @@
+test_that("can get chats/data from completed request", {
+  chat <- chat_openai_test()
+
+  prompts <- list(
+    "What's the capital of Iowa?",
+    "What's the capital of New York?",
+    "What's the capital of California?",
+    "What's the capital of Texas?"
+  )
+  chats <- batch_chat(
+    chat,
+    prompts,
+    path = test_path("batch/state-capitals.json")
+  )
+  expect_length(chats, 4)
+
+  type_state <- type_object(name = type_string("State name"))
+  data <- batch_chat_structured(
+    chat,
+    prompts,
+    path = test_path("batch/state-name.json"),
+    type = type_state
+  )
+  expect_equal(nrow(data), 4)
+})
+
+test_that("errors if chat/provider/prompts don't match previous run", {
+  chat <- chat_anthropic_test(system_prompt = "Be cool")
+  prompts <- list("What's the capital of Iowa?")
+  path <- test_path("batch/state-capitals.json")
```

Comment:
Instead of using the file from the previous test, you could store a small JSON with enough state information to trigger the hash checks.